### PR TITLE
Eclipse setup

### PR DESCRIPTION
**Pull request by weber-s**
PR of issue https://github.com/weber-s/apli_dash/issues/4 . branch development and Eclipse_setup has diverged. I'm working on it.

### CODE REVIEW

**Comment by weber-s** [comment]
Ok. Some big change were apply.

# Directory setup

The directory structure has to be as follows:

```
├── INSTALLATION.md
├── README.md
├── manage.py
├── pmall
│   ├── __init__.py
│   ├── __pycache__
│   ├── local_settings.py
│   ├── settings.py
│   ├── templates
│   ├── urls.py
│   ├── views.py
│   └── wsgi.py
├── pmall_app
│   ├── __init__.py
│   ├── __pycache__
│   ├── admin.py
│   ├── apli_bdd_aerosols.py
│   ├── apps.py
│   ├── migrations
│   ├── models.py
│   ├── server.py
│   ├── static
│   ├── tests.py
│   ├── urls.py
│   └── views.py
├── requirements.txt
└── templates
    ├── base.html
    ├── home.html
    └── registration
```

I move all the file in their respective directories.

Also, a file called `local_settings.py` shoudl go into the `pmall` folder, and be filled with something like that: 

```python
import os

# Build paths inside the project like this: os.path.join(BASE_DIR, ...)
BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))

SECRET_KEY = '&oupn^)a!at0fh3auyu+d#rfr)&z#cpdex5&o6^0m&887wec4y'

DEBUG = True

ALLOWED_HOSTS = ['*']

DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.sqlite3',
        'NAME': os.path.join(BASE_DIR, 'db_app.sqlite3'), # or whatever name (db_users.sqlite3 for instance)
    }
}

BDDPM = "path/to/the/PM/BDD/bdd.sqlite""
```

There is no more `app_bdd_settings.py`

# Templates/style

Thanks for the better style of apli_bdd_aerosols.py ! 

However, 

- there is a header entitled `place for navbar etc` ? Why ?
- the home template is still a default one with latin word and stuff like that...
- I changed a bit the layout of the app, based on new stuff present in the development branch

# To sum up

- I commit my change in the branch Eclipse_setup. You will need a `git pull` before re-working on it
- As I did a partial merge of development into Eclipse_setup, you will need to install new dependencies with `pip3 install -r requirements.txt` (support for Tabs).
- I remove the tracking of all the stuff related to a specific IDE and the tracking of the DB.
- Question: do I wait the whole layout of the site to be OK before merging or do I a partial merge (for apli_bdd_aerosols.py` ?

**Comment by weber-s** [comment]
By the way,

Lot of differences are due to handling end-of-line differently between our two OS... So if you want to see the "real" change, you have to add "?w=1" at the end of the url:

https://github.com/weber-s/apli_dash/pull/6/files?w=1

**Comment by MBoidot** [comment]
# Directory setup

Thanks for reorganizing.
db_user.sqlite3 looks fine for the user database.
db_data.sqlite3 ok for the data db


- I don't see the local_settings.py file even after I pulled. Shall I create it and push ?
- Where does the secret key come from ? 

# Styling/templates

- I just got the things to work. Now we can fill the templates with whatever is needed. I'll finish the styling after the branches are merged. (navbar/latin text/etc).
- you can go on for mergeing whatever is possible.

**Comment by weber-s** [comment]
Hi,

The local_settings.py has to be writen by you and not tracked by git. It is a specific file related to your configuration (what's the name of the DB, where it is store in your computer, do you want to print debug info, etc.). The Secret key is related to security issue when dealing with cryptographic stuff and is a random generated string. We don't really need one for know, but it has to be kept secret. That's why the local_settings.py is not tracked by git in production: the secret key is different for you, me, and production.

Ioups for the user database... I forgot that it will be deleted for you with the pull... To recreate one:

```python
python3 manage.py migrate
```
and then recreate a user with
```python
python3 manage.py createsuperuser
```

Concerning the template: I am not a fan of merging things in production that are still work-in-progress features. All that go in the master/development branch should be the state of what we want in production. For instance, in the home page right now, we don't see the link to the app, which is embarrassing.

I move them in a new branch `styling_template`. Doing so allowed to apply your change in apli_bdd_aerosols :) 

I don't understand

> It seems, that the dash_html_components doesn't have all the styling available, like span, navbar and other stuff, so I did as I could.

dash_html_components has a Nav and Span class (see [here](https://github.com/plotly/dash-html-components/tree/master/src/components)).

**Comment by MBoidot** [comment]
OK, i'll let you delete the eclipse_setup branch and continue working on the styling_template one.
